linetools/isgm/io.py

```diff
@@ -41,3 +50,67 @@ def abssys_from_json(filename):
 
     # Return
     return abs_sys
+
+
+def read_joebvp(filename, coord, llist=None):
+    """
+    Parameters
+    ----------
+    filename : str
+      joeB VP filename
+    coord : SkyCoord
+    llist : LineList, optional
+
+    Returns
+    -------
+    comps : list
+      list of AbsComponent objects
+    """
+    # init
+    if llist is None:
+        llist = LineList('ISM')
+    comps = []
+    # Read
+    vp_data = Table.read(filename, format='ascii')
+    nflags, nidx = np.unique(vp_data['nflag'], return_index=True)
+
+    # Subset by nflag; Build components
+    for nflag in nflags:
```

Comment:
I think we shouldn't pick out components by the 'nflag' values.  I see the logic and while this will work in many cases, 'nflag' can also take on values 0 (let parameter vary without being tied to other lines) or 1 (fix parameter value).  My approach chooses components based on 'zsys' and then 'trans' (which should probably be called 'species', but anyway...).  However, this isn't perfect either, because one may already have declared multiple components with the same systemic redshift but different velocities during the fit.  The best approach might involve employing the clustering analysis to group different transitions of the same species and redshift by velocity.

Reply:
egads, I worry about the clustering.

Let me try zsys + trans.